vm/core/hash.go

```diff
@@ -1,10 +1,11 @@
 package core
 
 import (
-	"github.com/spacemeshos/go-scale"
+	"github.com/ChainSafe/gossamer/pkg/scale"
```

Comment:
Because it's harder to work with. It requires pre-generated scale types. With gossamer, you just pass in the data structure instance and get its serialization.

Reply:
We already use `go-scale` extensively and it's not been a problem so far. IMHO, consistency (using the same tool for encoding) is important. Probably the problem would be that go-scale by default encodes integers in the compact way. It supports non-compact encoding, but I'm not sure if it has a struct tag to switch to non-compact - perhaps this would need to be added. Or we could switch to compact encoding on the VM side.